pkg/apis/certmanager/v1beta1/types_certificate.go

```diff
@@ -206,6 +179,34 @@ type CertificatePrivateKey struct {
 	// Default is 'Never' for backward compatibility.
 	// +optional
 	RotationPolicy PrivateKeyRotationPolicy `json:"rotationPolicy,omitempty"`
+
+	// The private key cryptography standards (PKCS) encoding for this
+	// certificate's private key to be encoded in.
+	// If provided, allowed values are "pkcs1" and "pkcs8" standing for PKCS#1
+	// and PKCS#8, respectively.
+	// Defaults to PKCS#1 if not specified.
+	Encoding PrivateKeyEncoding `json:"encoding,omitempty"`
```

Comment:
Shouldn't these be marked `+optional` and therefore shouldn't they be string pointers so that you can distinguish between the user supplying an invalid empty string value or no value ?

Reply:
We don't need to distinguish between not set and empty though - if the field is empty, it will be defaulted (albeit in the controller right now, for the reasons described in https://github.com/jetstack/cert-manager/pull/3038#discussion_r448964935).

`omitempty` works fine for non-struct non-pointer types - `PrivateKeyEncoding` is a `string`, so this has the desired behaviour 😄 once we have defaulting, users will also be able to *rely* on this being non-empty too.